File: ruby/2021/day1/day1.rb
data = File.new("day1.in").read().split("\n").map { |num| num.to_i }

part_1, part_2 = 0, 0

data[0..-2].zip(data[1..]).each do |first, second|
    part_1 += 1 unless first >= second
end

puts part_1

data[0..-4].zip(data[1..-3], data[2..-2], data[3..]).each do |group|
    part_2 += 1 unless group[..-2].sum >= group[1..].sum    
end

puts part_2

File: python/2021/day1/day1.py
#!/usr/bin/env python3
#import solution

with open('./day1.in', 'r') as file:
    data = list(map(int, file.read().splitlines()))

count1, count2 = (0, 0)

for x, y in zip(data[:-1], data[1:]):
    if y > x: count1 += 1

# does this even count as 'pythonic' any more...
for x, y in zip(map(sum, zip(data[:-3], data[1:-2], data[2:-1])), map(sum, zip(data[1:-2], data[2:-1], data[3:]))):
    if y > x: count2 += 1

print(count1, count2)